### PR TITLE
Export unique UniProt IDs in Atlas as Atlas search links

### PR DESCRIPTION
**Pull request by irisdianauy**
This adds a feature to export unique UniProt IDs in Atlas and their corresponding Atlas search links.  This currently only works for GXA.

### CODE REVIEW

File: bin/export_atlas_uniprot.sh [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+# 2. Generates an Atlas search linnk for each UniProt ID
+# 3. Writes everything to file
+
+# Caveats:
+# 1. Only works for bulk-analytics for now. There seems to be no  in scxa solr (???) still waiting for reply from devs
+# 2. Only gets 100 unique uniprot IDs...seems odd.  Still investigating this.
+
+
+atlasUniqueUniprotUrl="http://${SOLR_HOST}/solr/bulk-analytics-v1/select?facet.field=keyword_uniprot&facet=on&q=*:*&rows=0&start=0"
+ATLAS_UNIPROT_EXPORT_FILE=${$1:-$ATLAS_UNIPROT_EXPORT_FILE}
```
Comment (anilthanki) [inline bin/export_atlas_uniprot.sh]: What if `$1` is not defined and `$ATLAS_UNIPROT_EXPORT_FILE` not defined as well..

Generally here a default hardcoded filename given

File: bin/export_atlas_uniprot.sh [hunk 2]
```diff
@@ -0,0 +1,25 @@
+#!/usr/bin/env bash
+
+# What this script does:
+# 1. Gets all unique UniProt IDs in Atlas bulk-analytics collection
+# 2. Generates an Atlas search linnk for each UniProt ID
+# 3. Writes everything to file
+
+# Caveats:
+# 1. Only works for bulk-analytics for now. There seems to be no  in scxa solr (???) still waiting for reply from devs
```
Comment (anilthanki) [inline bin/export_atlas_uniprot.sh:9]: there is no scxa in dev solr.. we lost it hen we lost solr completely, but for testing try using public/fallback or dev on solr8

File: bin/export_atlas_uniprot.sh [hunk 3]
```diff
@@ -0,0 +1,25 @@
+#!/usr/bin/env bash
+
+# What this script does:
+# 1. Gets all unique UniProt IDs in Atlas bulk-analytics collection
+# 2. Generates an Atlas search linnk for each UniProt ID
```
Comment (anilthanki) [inline bin/export_atlas_uniprot.sh]:
```suggestion
# 2. Generates an Atlas search link for each UniProt ID
```

File: bin/export_atlas_uniprot.sh [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+
+# Caveats:
+# 1. Only works for bulk-analytics for now. There seems to be no  in scxa solr (???) still waiting for reply from devs
+# 2. Only gets 100 unique uniprot IDs...seems odd.  Still investigating this.
+
+
+atlasUniqueUniprotUrl="http://${SOLR_HOST}/solr/bulk-analytics-v1/select?facet.field=keyword_uniprot&facet=on&q=*:*&rows=0&start=0"
+ATLAS_UNIPROT_EXPORT_FILE=${$1:-$ATLAS_UNIPROT_EXPORT_FILE}
+
+rm -f $ATLAS_UNIPROT_EXPORT_FILE
```
Comment (pmb59) [inline bin/export_atlas_uniprot.sh]: I'd recommend to keep older exports, and add the date/time to the filename

File: bin/export_atlas_uniprot.sh (hunk 1 above)
Comment (irisdianauy) [reply bin/export_atlas_uniprot.sh]: Thanks Anil.  $ATLAS_UNIPROT_EXPORT_FILE is already defined in our production env (added when I tested this).

File: bin/export_atlas_uniprot.sh [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+
+# What this script does:
+# 1. Gets all unique UniProt IDs in Atlas bulk-analytics collection
+# 2. Generates an Atlas search linnk for each UniProt ID
+# 3. Writes everything to file
+
+# Caveats:
+# 1. Only works for bulk-analytics for now. There seems to be no  in scxa solr (???) still waiting for reply from devs
+# 2. Only gets 100 unique uniprot IDs...seems odd.  Still investigating this.
+
```
Comment (pmb59) [inline bin/export_atlas_uniprot.sh:15]:
```suggestion
# Check that relevant env vars are set
[ -z ${SOLR_HOST+x} ] && echo "Env var SOLR_HOST needs to be defined." && exit 1
[ -z ${SOLR_USER+x} ] && echo "Env var SOLR_USER needs to be defined." && exit 1
[ -z ${SOLR_PASS+x} ] && echo "Env var SOLR_PASS needs to be defined." && exit 1

command -v jq &>/dev/null || { echo "jq is not installed."; exit 1; }

```

File: bin/export_atlas_uniprot.sh (hunk 4 above)
Comment (irisdianauy) [reply bin/export_atlas_uniprot.sh]: Done in cd9dc4015b093a0be4c3eff7a730167b87a5fff1.

File: bin/export_atlas_uniprot.sh (hunk 1 above)
Comment (irisdianauy) [reply bin/export_atlas_uniprot.sh]: Done in 14a35a1dbc67eb413208241473b93f3456bfe6ed.

**Review by pmb59 — APPROVED** [review]
LGTM @irisdianauy
I'd suggest you merge when you are satisfied of the export in the production environment, i.e.
- export is complete with all UniProt IDs
- EMICS accepts the export format and has access to it
- we've decided for file option instead of the REST API

**Review by anilthanki — APPROVED** [review]
LGTM

**Comment by irisdianauy** [comment]
Thanks, @anilthanki and @pmb59.  Will wait for some inputs before merging this.

**Comment by irisdianauy** [comment]
Merging this now, as the ff have been satisfied:
>     * export is complete with all UniProt IDs
>     * EMICSS accepts the export format and has access to it
>     * we've decided for file option instead of the REST API
This script has been tested in codon.  File $ATLAS_UNIPROT_EXPORT_FILE exists and has been shared with EMICSS.

**Pull request merged**